core/domain/skill_services.py

```diff
@@ -277,6 +280,190 @@ def save_new_skill(committer_id, skill):
         }])
 
 
+def apply_change_list(skill_id, change_list):
+    """Applies a changelist to a skill and returns the result.
+
+    Args:
+        skill_id: str. ID of the given skill.
+        change_list: list(dict). A change list to be applied to the given
+            skill. Each entry in change_list is a dict that represents a
+            SkillChange object.
+
+    Returns:
+      Skill. The resulting skill domain object.
+    """
+    skill = get_skill_by_id(skill_id)
+    try:
+        changes = [skill_domain.SkillChange(change_dict)
+                   for change_dict in change_list]
+
+        for change in changes:
+            if change.cmd == skill_domain.CMD_UPDATE_SKILL_PROPERTY:
+                if (change.property_name ==
+                        skill_domain.SKILL_PROPERTY_DESCRIPTION):
+                    skill.update_description(change.new_value)
+                elif (change.property_name ==
+                      skill_domain.SKILL_PROPERTY_LANGUAGE_CODE):
+                    skill.update_language_code(change.new_value)
+            elif change.cmd == skill_domain.CMD_UPDATE_SKILL_CONTENTS_PROPERTY:
+                if (change.property_name ==
+                        skill_domain.SKILL_CONTENTS_PROPERTY_EXPLANATION):
+                    skill.update_explanation(change.new_value)
+                elif (change.property_name ==
+                      skill_domain.SKILL_CONTENTS_PROPERTY_WORKED_EXAMPLES):
+                    skill.update_worked_examples(change.new_value)
+            elif change.cmd == skill_domain.CMD_ADD_SKILL_MISCONCEPTION:
+                skill.add_misconception(change.misconception_id)
+            elif change.cmd == skill_domain.CMD_DELETE_SKILL_MISCONCEPTION:
+                skill.delete_misconception(change.misconception_id)
+            elif (change.cmd ==
+                  skill_domain.CMD_UPDATE_SKILL_MISCONCEPTIONS_PROPERTY):
+                if (change.property_name ==
+                        skill_domain.SKILL_MISCONCEPTIONS_PROPERTY_NAME):
+                    skill.update_misconception_name(
+                        change.misconception_id, change.new_value)
+                elif (change.property_name ==
+                      skill_domain.SKILL_MISCONCEPTIONS_PROPERTY_NOTES):
+                    skill.update_misconception_notes(
+                        change.misconception_id, change.new_value)
+                elif (change.property_name ==
+                      skill_domain.SKILL_MISCONCEPTIONS_PROPERTY_FEEDBACK):
+                    skill.update_misconception_feedback(
+                        change.misconception_id, change.new_value)
+            elif (change.cmd ==
+                  skill_domain.CMD_MIGRATE_CONTENTS_SCHEMA_TO_LATEST_VERSION
+                  or change.cmd ==
+                  skill_domain.CMD_MIGRATE_MISCONCEPTIONS_SCHEMA_TO_LATEST_VERSION): # pylint: disable=line-too-long
+                # Loading the skill model from the datastore into a
```

Comment:
Note that (later?) we should also have a one-off job that runs through all models and bulk-updates them to the new version, similar to ExplorationMigrationOneOffJob.

Reply:
Ok, will do that in a later PR.